cmd/zarb/tx/bond_tx.go

```diff
@@ -112,10 +110,40 @@ func BondTx() func(c *cli.Cmd) {
 				return
 			}
 
+			//sign transaction
+			if *keyFileOpt == "" {
+				cmd.PrintWarnMsg("Please specify a key file to sign.")
+				c.PrintHelp()
+				return
+			}
+			if *authOpt == "" {
+				auth = cmd.PromptPassphrase("Passphrase: ", false)
+			} else {
+				auth = *authOpt
+			}
+
+			//RPC
+			seq = *seqOpt
+			if *seqOpt == 0 {
+				seq, err = util.GetSequence(promptRPCEndpoint(*grpcOpt), bonder)
+				if err != nil {
+					cmd.PrintErrorMsg("Couldn't retrieve Sequence number from RPC Server: %v", err)
+					return
+				}
+			}
+
+			stamp, err = crypto.HashFromString(*stampOpt)
+			if err != nil {
+				stamp, err = util.GetStamp(promptRPCEndpoint(*grpcOpt))
+				if err != nil {
+					cmd.PrintErrorMsg("Couldn't retrieve stamp from RPC Server: %v", err)
+					return
+				}
+			}
```

Comment:
This part has a potential problem. User may enter invalid hash. So we ignore it and query from the node. 
Maybe we can rewrite it like this:
```
			
			if stampOpt == nil {
                stamp, err = crypto.HashFromString(*stampOpt)
                if err != nil {
					cmd.PrintErrorMsg("Stamp is not valid: %v", err)
					return
				}
           } else { 
				stamp, err = util.GetStamp(promptRPCEndpoint(*grpcOpt))
				if err != nil {
					cmd.PrintErrorMsg("Couldn't retrieve stamp from RPC Server: %v", err)
					return
				}
			}
```